src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -1911,7 +1911,7 @@ private Matches() {}
       }
 
       final Match<Unit> unitIsOwnedByAndNotDisabled = Match.allOf(
-          Matches.unitIsOwnedBy(unit.getOwner()), Matches.unitIsNotDisabled());
+          Matches.isUnitAllied(unit.getOwner(), data), Matches.unitIsNotDisabled());
```

Comment:
Would it be appropriate to update the corresponding unit test to reflect this new requirement?

Reply:
Yeah, updated the unit test.